### DISCUSSION TITLE
Client side caching with getServerSideProps

### DISCUSSION THREAD

**Discussion by nitsky** [vercel/next.js#11295] [Help]
Hi, right now with `getInitialProps` I can avoid refetching data that is used on multiple pages like so:

```js
MyPage.getInitialProps = async () => ({
  bigBlob: (isClient && bigBlobCache.get()) || await fetchBigBlob(),
})
```

This means that `bigBlob` will be fetched during server render and in the first client navigation to a page that uses `bigBlob`, but not in subsequent client navigations to pages that also use `bigBlob`. Is there a way to optimize data fetching in this way using the new `getServerSideProps`? Client side data fetching would work, but then if the user does a client navigation to the first page that uses `bigBlob`, they would have to see a flash of a loading indicator.

**Comment by izziaraffaele** [comment]
You can load data in `getStaticProps` and it will be fetched only once at build-time

```javascript
export async function getStaticProps() {
  return {
    props: { 
      bigBlob: await fetchBigBlob(),
    }
  }
}
```

**Comment by nitsky** [comment]
@izziaraffaele The data is user generated, confidential, and unavailable at build it. It is not eligible for static rendering.

**Comment by elbotho** [comment]
I did not get there yet, but with [Shallow Routing ](https://nextjs.org/docs/routing/shallow-routing) you can bypass the data fetching calls ( `getServerSideProps`, `getStaticProps`, and `getInitialProps`).
So you might use shallow routing and use the route change event to check if the data is in the cache already.

**Reply by barthuijgen** [reply]
It's a shame there's no good solution to this. Trying to use next.js for more then a simple blog often falls short of these types of "edge cases" as the maintainers would say. It's really not and edge case though IMO, it's just a basic performance optimisation that next.js should be able to easily support without writing ugly hacks and bypassing potentionally good features like `getServerSideProps`. 

I'm also working on a highly dynamic website making use of `getServerSideProps` and it's just not a great experience to optimise.

**Reply by C-Sinclair** [reply]
I agree with the above, it would be really cool if there was a story around this. 
For instance I have some logic for authenticating a user in `getServerSideProps` with one third party, and then using the result to query another service. This would be really nice if there was a way of avoiding subsequent `getServerSideProps` calls when the route changes. Shallow Routing is cool, but only works for change of query params on the same page. What if this was possible across pages?
